### PR TITLE
Use Django for accounts and persistence

### LINKED ISSUE #7179: Replace Firebase with our own Django Auth server B:180
### Description
We have been wanting to drop Firebase for a while now and the Django auth server is in good shape, we're missing the implementation in Unitystation side.

To resolve this issue, the contributor must:
- Be able to read the host of the api from a config file, making possible for others to host their own auth servers.
- Write a wrapper for WebRequest to communicate with the Django auth server, being able to:
    - Login using username and password, receiving a token.
    - Login and verify user using a username and token.
    - Create a new account.
    - Trigger a request to change password.
    - Trigger a request for "forgot password".
    - Leave room for new requirements as the final implementation is still wip.

```
- [POST]{"id", "username", "email", "password"} http://api.register
- [POST]{"email", "password" || "token"} http://api.login
- [POST]{"email", "password" || "token"} http://api.logout
- [POST]{"email"} http://api.forgot-password
- [POST]{"email", "password", "newPassword" || "token", "newPassword"} http://api.change-password
- [GET]{"id"} http://api.users/<id>/
- [GET]{"id", "index"} http://api.characters/<id>/<index>
```

### PR DESCRIPTION
**Pull request by NoooneyDude**
WIP django PR for as a reminder / motivation / early feedback. Plenty to clean up.

- Closes #7179.

CL: [Improvement] Use Django for accounts and persistence.

### CODE REVIEW

**Comment by corp-0** [comment]
image
 nice

File: UnityProject/Assets/Scripts/Core/Account/Account.cs
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Account/Account.cs]: did I already do this on django's side? I don't remember

File: UnityProject/Assets/Scripts/Core/Database/PersistenceModels.cs
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Database/PersistenceModels.cs:6]: for moderation

**Comment by MaxIsJoe** [comment]
Fixed merge conflicts on this made it up to date

**Comment by corp-0** [comment]
I can't see the changes. It would be great if we could set a flag somewhere so it uses Django or Firebase. Then we could merge this without fear of not being able to play and further PRs would be a lot smaller

**Comment by MaxIsJoe** [comment]
> I can't see the changes. It would be great if we could set a flag somewhere so it uses Django or Firebase. Then we could merge this without fear of not being able to play and further PRs would be a lot smaller

Firebase is completely removed from this PR.

**Comment by corp-0** [comment]
Which is bad because then we can't safely merge this without giving it exhaustive testing before, prolonging the merge date and making it harder to maintain up to date

**Comment by MaxIsJoe** [comment]
> Which is bad because then we can't safely merge this without giving it exhaustive testing before, prolonging the merge date and making it harder to maintain up to date

I say no, simply look at HealthV1 and HealthV2 and all the issues related to them existing at the same time.
Deprecated features should not be left in the project at all, as they become a hasel to maintain, while everyone else ignores the better alternative because the old method still exists.

Firebase has been such an issue, that it has started negatively affecting both players and developers. Under no circumstances should it be left in the project after we move over to using django.

File: UnityProject/Assets/Scripts/Systems/Lobby/CharacterSheet.cs
Comment (corp-0) [inline UnityProject/Assets/Scripts/Systems/Lobby/CharacterSheet.cs]: Character IDs on the backend are just an integer

**Review by MaxIsJoe — DISMISSED** [review]
Merge this once Bod works on the safe urls list thingy

**Comment by corp-0** [comment]
also change the url in config to prod-api

File: UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs]: @MaxIsJoe 
wasn't this exposed as a config already?

Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs]:
```suggestion
		public static string Host = GameManager.Instance.AccountAPIHost;
```

**Review by corp-0 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs]: it was, but something reverted it I believe; or this is an entirely different class that manually writes its own host thing

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Database/PersistenceServer.cs]: it is the later case. Already fixed it

**Review by MaxIsJoe — DISMISSED** [review]

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**
